src/main/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/service/NeedService.kt

```diff
@@ -71,4 +73,26 @@ class NeedService(
   fun hasNeed(prisonNumber: String): Boolean = hasActiveSANNeed(prisonNumber, includingALN = false) ||
     hasALNScreenerNeed(prisonNumber) ||
     hasLDDNeed(prisonNumber)
+
+  fun getNeedSources(prisonNumber: String): Set<NeedSource> {
+    val challenges = challengeRepository.findAllByPrisonNumber(prisonNumber)
+    val conditions = conditionRepository.findAllByPrisonNumber(prisonNumber)
+
+    return buildSet {
```

Comment:
Phwoar! `buildSet` and `add` .... that's some trick kotlin right there 😁 
I didn't know about those - they look nice 👍 

Every day's a school day 😁 👍